### PR TITLE
Send metrics and events to Azure Application Insights

### PR DESCRIPTION
**Pull request by lgonczar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RPE-114

### Change description ###

Send events and metrics to Application Insights

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pbis/servicebus/MessageQueueProcessor.java [hunk 1]
```diff
@@ -77,12 +88,15 @@ private void processMessages(IServiceBusClient serviceBusClient) {
             MessageProcessingResult processingResult = processMessage(message);
             updateMessageInSubscription(message, processingResult, serviceBusClient);
             logProcessingResult(processingResult, message);
+            sendTelemetryDataForMessage(processingResult);
```
Comment (KarolJastrzebski) [inline src/main/java/uk/gov/hmcts/reform/pbis/servicebus/MessageQueueProcessor.java]: It looks like `MessageQueueProcessor` emits events that trigger different behaviours (update message, log result, send telemetry) that do not belong directly to the processor itself. These concerns are separate and could be handled in event listeners.

File: src/main/java/uk/gov/hmcts/reform/pbis/servicebus/MessageQueueProcessor.java [hunk 2]
```diff
@@ -141,6 +155,27 @@ private void logProcessingResult(MessageProcessingResult processingResult, IMess
...
+            case SUCCESS:
+                telemetryClient.trackEvent(EMAIL_SENT);
+                break;
+            case ERROR:
+                telemetryClient.trackEvent(MESSAGE_PROCESSING_ERROR);
+                break;
+            case UNPROCESSABLE_MESSAGE:
+                telemetryClient.trackEvent(MESSAGE_REJECTED);
+                break;
+            default:
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/pbis/servicebus/MessageQueueProcessor.java]: may be worth logging something here.

File: src/main/java/uk/gov/hmcts/reform/pbis/telemetry/MetricNames.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.hmcts.reform.pbis.telemetry;
+
+
+public class MetricNames {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/pbis/telemetry/MetricNames.java]: perhaps `CounterNames`?

File: src/main/java/uk/gov/hmcts/reform/pbis/servicebus/MessageQueueProcessor.java (hunk 1 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/pbis/servicebus/MessageQueueProcessor.java]: keep it simple

File: src/main/java/uk/gov/hmcts/reform/pbis/servicebus/MessageQueueProcessor.java (hunk 2 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/pbis/servicebus/MessageQueueProcessor.java]: done

**Pull request merged**